gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -399,6 +503,14 @@ public void onAddFeatureBtnClick() {
     addFeatureButtonClicks.onNext(getCameraPosition().getValue().getTarget());
   }
 
+  public void onAddPolygonBtnClick() {
+    if (vertices.contains(getCameraPosition().getValue().getTarget())) {
+      updatePolygonDrawing(PolygonDrawing.COMPLETED);
+    }
```

Comment:
`getCameraPosition` invokes `checkPointNearVertex` which is also doing the `near first vertex` check. Please check again.

Reply:
moved to onCameraMove so that if the camera point is near to first vertex the polygon automatically completes.